complex_project/my_complex.h

```diff
@@ -1,85 +1,75 @@
-//
 
-#ifndef TEST1_MY_COMPLEX_H
-#define TEST1_MY_COMPLEX_H
+#pragma once
 
 #include <iostream>
+#include <cmath>
 
-class MyComplex
+struct MyComplex
 {
-private:
-    double real;
-    double imag;
-public:
-    MyComplex(): real(0.0), imag(0.0) {}
-    MyComplex(double real): real(real), imag(0.0) {}
-    MyComplex(double real, double imag): real(real), imag(imag) {}
-
-    // Copy constructor
-    MyComplex(const MyComplex&C)
-    {
-        real = C.real;
-        imag = C.imag;
+    MyComplex(double real = 0.0, double imag = 0.0): m_real(real), m_imag(imag) {}
+
+    MyComplex(const MyComplex& c){
+        m_real = c.m_real;
+        m_imag = c.m_imag;
     }
 
-    // operator =
-    MyComplex& operator=(const MyComplex& C)
-    {
-        real = C.real;
-        imag = C.imag;
+    MyComplex& operator=(const MyComplex& c){
+        m_real = c.m_real;
+        m_imag = c.m_imag;
         return *this;
     }
 
-    // operator +
-    MyComplex operator+(const MyComplex& C)
-    {
-        return MyComplex(real + C.real, imag + C.imag);
+    MyComplex operator+(const MyComplex& c) const {
+        return MyComplex(m_real + c.m_real, m_imag + c.m_imag);
     }
 
-    // operator -
-    MyComplex operator-(const MyComplex& C)
-    {
-        return MyComplex(real - C.real, imag - C.imag);
+    MyComplex operator-(const MyComplex& c) const {
+        return MyComplex(m_real - c.m_real, m_imag - c.m_imag);
     }
 
-    // operator *
-    MyComplex operator*(const MyComplex& C)
-    {
-        return MyComplex(real*C.real - imag*C.imag, real*C.imag + imag*C.real);
+    MyComplex operator*(const MyComplex& c) const {
+        return MyComplex(m_real * c.m_real - m_imag * c.m_imag, m_real * c.m_imag + m_imag * c.m_real);
     }
 
-    // operator +=
-    MyComplex& operator+=(const MyComplex& C)
-    {
-        real += C.real;
-        imag += C.imag;
+    MyComplex& operator+=(const MyComplex& c){
+        m_real += c.m_real;
+        m_imag += c.m_imag;
         return *this;
     }
 
-    // operator -=
-    MyComplex& operator-=(const MyComplex& C)
-    {
-        real -= C.real;
-        imag -= C.imag;
+    MyComplex& operator-=(const MyComplex& c){
+        m_real -= c.m_real;
+        m_imag -= c.m_imag;
         return *this;
     }
 
-    // operator *=
-    MyComplex& operator*=(const MyComplex& C)
-    {
-        real = real*C.real - imag*C.imag;
-        imag = real*C.imag + imag*C.real;
+    MyComplex& operator*=(const MyComplex& c){
+        double m_real_orig = m_real;
+        m_real = m_real_orig * c.m_real - m_imag * c.m_imag;
+        m_imag = m_real_orig * c.m_imag + m_imag * c.m_real;
         return *this;
     }
 
-    friend  std::ostream& operator<<(std::ostream& cout, const MyComplex& C)
-    {
-        cout << C.real << " + " << C.imag << "i"  << " ";
-        return cout;
+    bool operator==(const MyComplex& c) const {
+        if ( abs(m_real-c.m_real) > 0.0000001)
+            return false;
```

Comment:
We always enclose blocks in `{ ... }`:
```suggestion
        if (abs(m_real - c.m_real) > 0.0000001) {
            return false;
        }
```

Also, why did you choose this threshold? I was a bit surprised to see this behaviour.